tests/base/test_array.py

```diff
@@ -0,0 +1,467 @@
+        assert result.shape == (3,)
+
+
+class TestRhat:
+    def test_rhat_basic(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        result = array_stats.rhat(ary)
+        assert result.shape == ()
+        assert result > 0
+
+    @pytest.mark.parametrize("method", ["rank", "folded", "z_scale", "split", "identity"])
+    def test_rhat_methods(self, array_stats, method):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        result = array_stats.rhat(ary, method=method)
+        assert result.shape == ()
+        assert result > 0
+
+    def test_rhat_invalid_method(self, array_stats):
+        ary = np.random.randn(4, 100)
+        with pytest.raises(ValueError, match="must be one of"):
+            array_stats.rhat(ary, method="invalid")
+
+    def test_rhat_multiple_vars(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(3, 4, 100))
+        result = array_stats.rhat(ary)
+        assert result.shape == (3,)
+
+
+class TestRhatNested:
+    def test_rhat_nested_basic(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        superchain_ids = np.array([0, 0, 1, 1])
+        result = array_stats.rhat_nested(ary, superchain_ids)
+        assert result.shape == ()
+        assert result > 0
+
+    @pytest.mark.parametrize("method", ["rank", "folded", "z_scale", "split", "identity"])
+    def test_rhat_nested_methods(self, array_stats, method):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        superchain_ids = np.array([0, 0, 1, 1])
+        result = array_stats.rhat_nested(ary, superchain_ids, method=method)
+        assert result.shape == ()
+
+    def test_rhat_nested_invalid_method(self, array_stats):
+        ary = np.random.randn(4, 100)
+        superchain_ids = np.array([0, 0, 1, 1])
+        with pytest.raises(ValueError, match="must be one of"):
+            array_stats.rhat_nested(ary, superchain_ids, method="invalid")
+
+
+class TestMCSE:
+    def test_mcse_basic(self, array_stats):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        result = array_stats.mcse(ary)
+        assert result.shape == ()
+        assert result > 0
+
+    @pytest.mark.parametrize("method", ["mean", "sd", "median", "quantile"])
+    def test_mcse_methods(self, array_stats, method):
+        rng = np.random.default_rng(42)
+        ary = rng.normal(size=(4, 100))
+        prob = 0.95 if method == "quantile" else None
+        result = array_stats.mcse(ary, method=method, prob=prob)
+        assert result.shape == ()
+        assert result > 0
+
+    def test_mcse_invalid_method(self, array_stats):
+        ary = np.random.randn(4, 100)
+        with pytest.raises(ValueError, match="must be one of"):
+            array_stats.mcse(ary, method="invalid")
+
+
+class TestParetoFunctions:
```

Comment:
I would also add tests for ndarray input, with both integer and list/array input for `axis` argument (with the exception of pareto_min_ss which takes chain_axis and draw_axis so will probably look closer to the cases above)